### PR TITLE
[Features] Breadcrumbs Folders navigation

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

Breadcrumbs feature for Media Library folders & MediaLibraryInput modals

## How to test it

Navigate into nested folders in the Media Library
Use Breadcrumbs to navigate into previous folders

## Snapshot

image

image

### CODE REVIEW

**Review by MarionLemaire — APPROVED** [review]
Redirections tested on both Media Library section and Upload modal, on 10 nested levels of folders, using Getstarted : OK to me (to go to nested folder and to root).

Breadcrumbs display tested on Upload modal using Getstarted on 10 levels of nested folders ; OK.
image

Breadcrumbs display tested on Chrome using Getstarted on 10 levels of nested folders : OK.
image

Design QA done by @maevalienard : OK too.

File: packages/core/upload/admin/src/components/AssetDialog/BrowseStep/index.js [hunk 1]
```diff
@@ -88,48 +99,58 @@ export const BrowseStep = ({
...
-                </EndBlockActions>
-              )}
-            </Flex>
-          </Box>
+      {canRead && breadcrumbs?.length > 0 && (
+        <Box paddingTop={3}>
+          <Breadcrumbs
+            onChangeFolder={onChangeFolder}
+            as="nav"
+            label="hello"
```
Comment (gu-stav) [inline packages/core/upload/admin/src/components/AssetDialog/BrowseStep/index.js:150]: This seems to be leftover.

File: yarn.lock [hunk 2]
```diff
@@ -2640,7 +2640,7 @@
   resolved "https://registry.yarnpkg.com/@hutson/parse-repository-url/-/parse-repository-url-3.0.2.tgz#98c23c950a3d9b6c8f0daed06da6c3af06981340"
   integrity sha512-H9XAx3hc0BQHY6l+IFSWHDySypcXsvsuLhgYLUGywmJ5pswRVQJUHpOsobnLYp2ZUaUlKiKDrgWWhosOwAEM8Q==
 
-"@internationalized/number@^3.1.1":
+"@internationalized/number@^3.0.2", "@internationalized/number@^3.1.1":
```
Comment (gu-stav) [inline yarn.lock:2643]: This should not be committed. Could you please update the branch to the latest master and remove the change?

File: packages/core/upload/admin/src/utils/getFolderURL.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    {
+      ...queryParamsWithoutQ,
+      folder: folder?.id,
+    },
+    { encode: false }
+  );
+
+  // Search query will always fetch the same results
+  // we remove it here to allow navigating in a folder and see the result of this navigation
+  return `${pathname}${queryParamsString ? `?${queryParamsString}` : ''}`;
```
Comment (gu-stav) [inline packages/core/upload/admin/src/utils/getFolderURL.js:15]: I think you don't have to change it, but it would be much nicer to use the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL) for such as case.

File: packages/core/upload/admin/src/utils/getBreadcrumbDataML.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import getFolderURL from './getFolderURL';
+
+const getBreadcrumbDataML = (folder, { pathname, query }) => {
+  let data = [
+    {
+      id: null,
+      label: 'Media Library',
```
Comment (gu-stav) [inline packages/core/upload/admin/src/utils/getBreadcrumbDataML.js:7]: Is this translated somewhere?

File: packages/core/upload/admin/src/utils/getBreadcrumbDataCM.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+const getBreadcrumbDataML = folder => {
+  let data = [
+    {
+      id: null,
+      label: 'Media Library',
```
Comment (gu-stav) [inline packages/core/upload/admin/src/utils/getBreadcrumbDataCM.js:5]: Is this translated somewhere?

File: packages/core/upload/admin/src/pages/App/tests/MediaLibrary.test.js [hunk 6]
```diff
@@ -179,59 +164,39 @@ describe('Media library homepage', () => {
...
-          data: {},
-        });
-        useFolders.mockReturnValueOnce({
-          data: [],
-          isLoading: false,
-          error: null,
-        });
-        renderML();
-
-        expect(
```
Comment (gu-stav) [inline packages/core/upload/admin/src/pages/App/tests/MediaLibrary.test.js:226]: Why did you remove that test?

**Review by gu-stav — COMMENTED** [review]
Added a few minor comments: nothing major. I've tested it and think it works very well. Good job on that.

What I am not happy with is the initial view of the media library and the modal: I find "Media Library" repetitive in the ML and it makes the modal very noisy. Similar to the back button: what do you think about hiding the breadcrumbs, if `currentFolder === null`?

Screenshot 2022-08-04 at 11 27 00

Screenshot 2022-08-04 at 11 25 14

/cc @maevalienard

**Comment by maevalienard** [comment]
@gu-stav That would work to me, I didn't know it was technically feasible

File: packages/core/upload/admin/src/pages/App/tests/MediaLibrary.test.js (hunk 6 above)
Comment (ronronscelestes) [reply packages/core/upload/admin/src/pages/App/tests/MediaLibrary.test.js:226]: good question 🤔

Comment (ronronscelestes) [reply packages/core/upload/admin/src/pages/App/tests/MediaLibrary.test.js:226]: mmh I guess I thought this test didn't make sense - I will add it back and fix it (we want to check that select all checkbox isn't there not if There are no elements with the applied filters is or is not in the document)

File: packages/core/upload/admin/src/utils/getFolderURL.js (hunk 3 above)
Comment (ronronscelestes) [reply packages/core/upload/admin/src/utils/getFolderURL.js:15]: interesting, since we use this stringify format at multiple places within the app I would propose to add this to my backlog for another PR and fix it everywhere instead of just here, wdyt?

**Comment by ronronscelestes** [comment]
@gu-stav @maevalienard @MarionLemaire 
integrated: Breadcrumbs will not show if the user is at the root level of the Media Library

**Review by MarionLemaire — APPROVED** [review]
Tested on Gestarted using Chrome : 

For Media Libary section : when on root, breadcrumb is not displayed. As soon as you get to a nested folder, breadcrumb appears. Redirections are still OK. Tested on 10 levels of nesting.

Same thing for Upload modal.

>> Ready to release !

**Review by gu-stav — APPROVED** [review]
Looks good to me. Let's get it on the road! 🚢 

Btw: thanks for getting rid of the title truncation. That's a big win!
